cirq-google/cirq_google/engine/engine_client_test.py

```diff
@@ -1229,3 +1270,71 @@ def test_list_time_slots(client_constructor):
 
     client = EngineClient()
     assert client.list_time_slots('proj', 'processor0') == results
+
+
+@mock.patch.object(quantum, 'QuantumEngineServiceAsyncClient', autospec=True)
+def test_run_job_over_stream(client_constructor):
+    fake_client = setup_fake_quantum_run_stream_client(client_constructor)
+
+    code = any_pb2.Any()
+    run_context = any_pb2.Any()
+    labels = {'hello': 'world'}
+    client = EngineClient()
+    expected_result = quantum.QuantumResult(parent='projects/proj/programs/prog/jobs/job0')
+    mock_responses = [quantum.QuantumRunStreamResponse(message_id='0', result=expected_result)]
+    fake_client.add_responses_and_exceptions(mock_responses)
+
+    actual_result = client.run_job_over_stream(
+        'proj', 'prog', code, 'job0', ['processor0'], run_context, 10, 'A job', labels
+    ).result(timeout=5)
+    client.stop_stream()
+
+    assert actual_result == expected_result
+    assert fake_client.stream_request_count == 1
+
+
+@mock.patch.object(quantum, 'QuantumEngineServiceAsyncClient', autospec=True)
+def test_run_job_over_stream_cancellation(client_constructor):
+    fake_client = setup_fake_quantum_run_stream_client(client_constructor)
+
+    code = any_pb2.Any()
+    run_context = any_pb2.Any()
+    labels = {'hello': 'world'}
+    client = EngineClient()
+    expected_result = quantum.QuantumResult(parent='projects/proj/programs/prog/jobs/job0')
+    mock_responses = [quantum.QuantumRunStreamResponse(message_id='0', result=expected_result)]
```

Comment:
Consider inlining this in the following line since this value is never otherwise used and `add_responses_and_exceptions` makes the usage clear.

Reply:
In the updated implementation, I think leaving it separate makes fake client construction more clear.